crates/katana/rpc/rpc/tests/starknet.rs

```diff
@@ -172,5 +176,171 @@ async fn test_send_declare_and_deploy_legacy_contract() {
         class_hash
     );
 
-    sequencer.stop().expect("failed to stop sequencer");
+    drop(katana_runner)
+}
+
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    StarknetContract,
+    "tests/test_data/solidity/StarknetMessagingLocalCompiled.json"
+);
+
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    Contract1,
+    "tests/test_data/solidity/Contract1Compiled.json"
+);
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_messaging_l1_l2() {
+    // Prepare Anvil + Messaging Contracts
+    let anvil_runner = AnvilRunner::new().await.unwrap();
+
+    let contract_strk = StarknetContract::deploy(anvil_runner.provider()).await.unwrap();
+    let strk_address = contract_strk.address();
+
+    assert_eq!(
+        contract_strk.address(),
+        &Address::from_str("0x5fbdb2315678afecb367f032d93f642f64180aa3").unwrap()
+    );
+
+    let contract_c1 = Contract1::deploy(anvil_runner.provider(), *strk_address).await.unwrap();
+
+    assert_eq!(
+        contract_c1.address(),
+        &Address::from_str("0xe7f1725e7734ce288f8367e1bb143e90bb3f0512").unwrap()
+    );
+
+    // Prepare Katana + Messaging Contract
+    let messagin_config = json!({
+        "chain": "ethereum",
+        "rpc_url": anvil_runner.endpoint,
+        "contract_address": contract_strk.address().to_string(),
+        "sender_address": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
```

Comment:
not the scope of this PR but i think the sender address isn't really required for evm chains. i think we should have a chain-specific config types cc @glihm